anchore_engine/services/policy_engine/engine/policy/gates/vulnerabilities.py

```diff
@@ -218,10 +219,19 @@ def evaluate(self, image_obj, context):
 
                                     if is_fix_available and fix_timeallowed is not None:
                                         if fix_available_in:
-                                            if calendar.timegm(vulnerability_cpe.created_at.timetuple()) > fix_timeallowed:
+                                            fix_date = vulnerability_cpe.created_at
+
+                                            # VulnDB entries might have a solution_date which is more accurate than the CPE created_at field
+                                            if vulnerability_cpe.parent.vuln_metadata:
+                                                vulndb_solution_date = vulnerability_cpe.parent.vuln_metadata['solution_date']
```

Comment:
I'd like this to be a bit more protected against KeyErrors using:

```suggestion
                                                vulndb_solution_date = vulnerability_cpe.parent.vuln_metadata.get('solution_date')
```

Reply:
Good catch, thanks!